src/pages/en/guides/migrate-to-astro/from-gatsby.mdx

```diff
@@ -2,67 +2,576 @@
+
+<AstroJSXTabs>
+  <Fragment slot="jsx">
+    ```jsx title="component.jsx"
+    import * as React from "react"
+    import PropTypes from "prop-types"
+    import { useStaticQuery, graphql } from "gatsby"
+
+    import Header from "./header"
+    import Footer from "./footer"
+    import "./layout.css"
+
+    const Component = ({ message, children }) => {
+      const data = useStaticQuery(graphql`
+        query SiteTitleQuery {
+          site {
+            siteMetadata {
+              title
+            }
+          }
+        }
+      `)
+
+      if (message === "denied") {
+        return (
+            <p>You are not authorized to see this page.</p>
+        )
+      }
+
+      return (
+        <>
+          <Header siteTitle={data.site.siteMetadata.title} />
+          <div style={{ 
+            margin: `0 auto`,
+            backgroundColor: `#f4f4f4`, 
+            padding: `1em 1.5em`
+            }}>{message}</div>  
+          <div
+            style={{
+              margin: `0 auto`,
+              maxWidth: 960,
+              padding: `1.25rem`,
+            }}
+          >
+            <main>{children}</main>
+            </div>
+          
+          <Footer siteTitle={data.site.siteMetadata} />
+
+        </>
+      )
+    }
+
+    Component.propTypes = {
+      children: PropTypes.node.isRequired,
+    }
+
+    export default Component
+    ``` 
+  </Fragment>
+
+  <Fragment slot="astro">
+    ```astro title="component.astro"
+    ---
+    import Header from "./Header.astro"
+    import Footer from "./Footer.astro"
+    import "../styles/stylesheet.css"
+    import { site } from "../data/siteMetaData.js"
+    const { message } = Astro.props
+    ---
+    {message === "denied"
+      ? <p>You are not authorized to see this page.</p>
+
+      : <Header siteTitle={site.title} />
+        <div class="banner">{message}</div>  
+        <div style="margin: 0 auto; max-width: 960; padding: 1.25rem;">
+          <main>
+            <slot />
+          </main>
+        </div>
+        <Footer siteTitle={site.title} />
+    }
+
+    <style>
+      .banner { 
+        margin: 0 auto;
+        background-color: #f4f4f4; 
+        padding: 1em 1.5em;
+      }  
+    </style>
+    ```
+  </Fragment>
+</AstroJSXTabs>
+
+### Migrating Layout Files
+
+In Astro, your layout must include a full page shell (`<html>`, `<head>` and `<body>` tags) and a `<slot/>` instead of React's `{children}` prop. Your Gatsby `layout.js` and templates will not include these. 
+
+You can use the following code to provide these extra elements around an existing Gatsby layout file, or write your own:
+
+```astro title="src/layouts/Layout.astro" "slot"
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" type="image/svg+xml" href="/favicon.svg" />
+    <meta name="viewport" content="width=device-width" />
+    <title>Astro</title>
+  </head>
+  <body>
+    <slot />
```

Comment:
Worth adding some comment to the effect that the “existing Gatsby layout” would go here?

```suggestion
    <!-- Wrap the slot element with your existing layout templating -->
    <slot />
```